server/channels/api4/user.go

```diff
@@ -2262,6 +2264,38 @@ func attachDeviceId(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func setExtraSessionProps(c *Context, w http.ResponseWriter, r *http.Request) {
+	newProps := map[string]string{}
+	receivedProps := model.MapFromJSON(r.Body)
+
+	deviceNotificationsDisabled := receivedProps[model.SessionPropDeviceNotificationDisabled]
+	if deviceNotificationsDisabled != "" && deviceNotificationsDisabled != "false" && deviceNotificationsDisabled != "true" {
+		c.SetInvalidParam(model.SessionPropDeviceNotificationDisabled)
+		return
+	}
+	if deviceNotificationsDisabled != "" {
+		newProps[model.SessionPropDeviceNotificationDisabled] = deviceNotificationsDisabled
+	}
+
+	mobileVersion := receivedProps[model.SessionPropMobileVersion]
+	if mobileVersion != "" {
+		if _, err := semver.Parse(mobileVersion); err != nil {
+			c.SetInvalidParam(model.SessionPropMobileVersion)
+			return
+		}
+	}
+	if mobileVersion != "" {
+		newProps[model.SessionPropDeviceNotificationDisabled] = mobileVersion
+	}
```

Comment:
There is a typo here on setting the mobile version to the wrong prop. 

Also, nit: we can probably simplify these checks as `semver.Parse` should handle empty string. 
```suggestion
		
	if _, err := semver.Parse(mobileVersion); err != nil {
		c.SetInvalidParam(model.SessionPropMobileVersion)
		return
	}

	newProps[model.SessionPropMobileVersion] = mobileVersion
```

Reply:
Thanks for spotting the typo!

Regarding not checking for the empty string, the idea here is that if we are passing an empty string (or no string whatsoever) means that that is not the extra prop we are setting, but something else.

The general idea is that this eventually will verify a number of props (like desktop version, for example), and not all clients will send all the props.

Reply:
> Regarding not checking for the empty string, the idea here is that if we are passing an empty string (or no string whatsoever) means that that is not the extra prop we are setting, but something else.

Good point, makes sense